pkg/storage/sqlstorage/balances.go

```diff
@@ -0,0 +1,202 @@
+package sqlstorage
+
+import (
+	"context"
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/huandu/go-sqlbuilder"
+	"github.com/numary/go-libs/sharedapi"
+	"github.com/numary/ledger/pkg/core"
+	"github.com/numary/ledger/pkg/storage"
+	"github.com/pkg/errors"
+)
+
+func (s *Store) GetBalancesAccountsData(ctx context.Context, exec executor, q storage.BalancesQuery) (sharedapi.Cursor[core.AccountsBalances], error) {
+	sb := sqlbuilder.NewSelectBuilder()
+	switch s.Schema().Flavor() {
+	case sqlbuilder.PostgreSQL:
+		sb.Select("account", "array_agg((asset, input - output))")
+	case sqlbuilder.SQLite:
+		// we try to get the same format as array_agg from postgres : {"(USD,-12686)","(EUR,-250)"}
```

Comment:
Maybe you could implement a "array_agg" sqlite function to avoid the switch but it seems to works like this.

Reply:
yeah i thought about it. i was not really familiar with this option, so i chose to implement it in the service, but no problem to try and do it the other way if the need arises